File: prisma/refs/combination-target-sum.ts
export function combinationSum(nums: number[], target: number): number[][] {
  const res: number[][] = [];
  const cur: number[] = [];
  const bt = (start: number, rem: number) => {
    if (rem === 0) {
      res.push([...cur]);
      return;
    }
    if (rem < 0) return;
    for (let i = start; i < nums.length; i++) {
      cur.push(nums[i]);
      bt(i, rem - nums[i]);
      cur.pop();
    }
  };
  bt(0, target);
  return res;
}

File: prisma/refs/py/combination-target-sum.py
def combination_sum(nums, target):
    res, cur = [], []

    def bt(start, rem):
        if rem == 0:
            res.append(cur[:])
            return
        if rem < 0:
            return
        for i in range(start, len(nums)):
            cur.append(nums[i])
            bt(i, rem - nums[i])
            cur.pop()

    bt(0, target)
    return res
